core/domain/exp_jobs_one_off.py

```diff
@@ -831,3 +832,100 @@ def reduce(key, values):
+            return
+
+        try:
+            if exp_rights.status == rights_manager.ACTIVITY_STATUS_PRIVATE:
+                exploration.validate()
+            else:
+                exploration.validate(strict=True)
+        except Exception as e:
+            yield('Error %s when validating current exploration' % str(e),
+                  item.id)
+            return
+
+
+        states = exploration.states
+        new_states = {}
+        for state_name, state in states.iteritems():
+            state_dict = state.to_dict()
+            state_content_id_list = []
+
+            # Add state card's content id into the state_content_id_list.
+            state_content_id_list.append(state_dict['content']['content_id'])
+
+            # Add answer_groups content id into the state_content_id_list.
+            for answer_group in state_dict['interaction']['answer_groups']:
+                answer_feedback = answer_group['outcome']['feedback']
+                state_content_id_list.append(answer_feedback['content_id'])
+
+            # If present, add default_outcome content id into
+            # state_content_id_list.
+            default_outcome = state_dict['interaction']['default_outcome']
+            if default_outcome is not None:
+                state_content_id_list.append(
+                    default_outcome['feedback']['content_id'])
+
+            # Add hints content id into state_content_id_list.
+            for hint in state_dict['interaction']['hints']:
+                state_content_id_list.append(hint['hint_content']['content_id'])
+
+            # If present, add solution content id into state_content_id_list.
+            solution = state_dict['interaction']['solution']
+            if solution:
+                state_content_id_list.append(
+                    solution['explanation']['content_id'])
+
+            # Filter content_ids_to_audio_translations with unwanted
+            # content id.
+            citat = state_dict['content_ids_to_audio_translations']
+            extra_content_ids_in_citat = (
+                set(citat.keys()) - set(state_content_id_list))
+            for content_id in extra_content_ids_in_citat:
+                state_dict['content_ids_to_audio_translations'].pop(content_id)
+
+            # Create written_translations using the state_content_id_list.
+            state_dict['written_translations'] = {}
+            for content_id in state_content_id_list:
+                state_dict['written_translations'][content_id] = {}
+
+            if state_dict['written_translations'].keys() != (
+                    state_dict['content_ids_to_audio_translations'].keys()) != (
+                        state_content_id_list):
+                yield ('Error when validating exploration in dict.', item.id)
+
+            new_states[state_name] = state_domain.State.from_dict(state_dict)
+
+        exploration.states = new_states
+        try:
+            exploration.validate()
```

Comment:
You've converted the states to the new schema, so will the validation for the previous schema still work?

Reply:
Yeah, it will still work as the State.from_dict(new_state_dcit) will not use written_translation while creating the object.

Reply:
OK. Maybe add a note here to explain that, since otherwise it just looks like an omission in reasoning.

Also, I guess I'm not entirely sure why we're doing the check in that case -- unless the reasoning is that .validate() + the checks above = a full validation? If so, that's worth saying and I agree it's a good idea to check. 

But, if that's the case, don't we need more checks to "equal" the validation in #6175? I thought there were a few sets of validations pertaining to the content translations at different levels.

Reply:
I think the checks are equal as the internal parts (Writtent_translation and langauge_code) of written_translations are going to be empty, so we will just have to validate the content_ids.

Added comment for explanation.